modal.js

```diff
@@ -0,0 +1,144 @@
+  <div class="popup">
+    <div class="top-title">
+      <h2 class="title">${title}</h2>
+      <button id="closed-popup-${id}" class="closed-popup"> 
+      <img src="./sources/closeButton.png" alt=""> </button>  
+    </div>
+  
+    <ul class="job-title">
+      <li class="canopy-title">
+      ${jobDetails[0]}
+      </li>
+      <li class="dot-icon">
+        <img class="dot-icon-img" src="./sources/dot.png" alt="dot"/>
+        ${jobDetails[1]}
+      </li>
+      <li class="year-2015">
+        <img class="year-tonic" src="./sources/dot.png" alt="year-tonic"/>
+        ${jobDetails[2]}
+      </li>
+    </ul>
+
+  <img id="modal-image" src="${featuredimage}" alt="picture">
+
+
+   <div class="bottom-content">
+       <p class="popupparagraph2">${jobDescriptionDetails}</p>
+   
+       <div class="right-block">
+         <div class="techstackbuttons">
+           <ul class="popupresources" title="popup-resources">
+             <li><a  class="popupbutton" href="#">${myTechStack[0]}</a><li>
+             <li><a  class="popupbutton" href="#">${myTechStack[1]}</a></li>
+             <li><a  class="popupbutton" href="#">${myTechStack[2]}</a></li>           
+           </ul>
+         </div>
+         
+         <div class="popupaction">
+           <button class="popupseeLiveButton">
+             <a href="${liveDemoButton}"> 
+             See Live
+             <img src="./sources/popupaction.png" alt="see image live" />
+             </a> 
+             </button> 
+           <button class="popupseeSourceButton " > 
+           <a href="${sourceLinkButton}"> 
+           See Source 
+             <img src="./sources/popupgithub.png" alt="github source" />
+             </a>
+             </button>
+         </div>
+       </div>
+     </div>
+   </div>
+</section>
+  `;
+}
+
+let modalCard = '';
+
+projectsDetails.forEach((index) => {
+  const modalDetails = generateModal(
+    index.id,
+    index.title,
+    index.jobDetails,
+    index.featuredimage,
+    index.jobDescriptionDetails,
+    index.myTechStack,
+    index.liveDemoButton,
+    index.sourceLinkButton,
+  );
+  modalCard += modalDetails;
+});
+
+// execute popup
+document.getElementById('modal-popup').innerHTML = modalCard;
+
+// document.getElementById(`openpop-${id}`).addEventListener('click', () => {
+//   document.getElementById(`popup-project-${id}`).classList.add('visible');
+// });
+// // close event
+// document.getElementById(`close-${id}`).addEventListener('click', () => {
+//   document.getElementById(`popup-project-${id}`).classList.remove('visible');
+
+projectsDetails.forEach((item) => {
+  const { id } = item;
+  document.getElementById(`modalButton-${id}`).addEventListener('click', () => {
+    document.getElementById(`popup-project-${id}`).classList.add('active');
+  });
+  document.getElementById(`closed-popup-${id}`).addEventListener('click', () => {
+    document.getElementById(`popup-project-${id}`).classList.remove('active');
+  });
+});
```

Comment:
[OPTIONAL] Well done overall but using just one modal will also make this easier 🥡  as you'd be able to target the same `close` element for every popup, because you'll only have one. 😉